hedera-node/hedera-app/src/test/java/com/hedera/node/app/util/HashTreeManagerTest.java

```diff
@@ -0,0 +1,168 @@
+
+        tree.addElement("leaf4");
+        String newRootHash = tree.getTreeRootAsString();
+        // Root hash should change as new leaves are added
+        assertThat(rootHash).as("Checking that hashes are not equal").isNotEqualTo(newRootHash);
+    }
+
+    @Test
+    public void testHashBlockItem() throws NoSuchAlgorithmException {
+        final String expectedHash =
+                "38b060a751ac96384cd9327eb1b1e36a21fdb71114be07434c0cc7bf63f6e1da274edebfe76f65fbd51ad2f14898b95b";
+        var rootHash = testAcceptBlockItem(blockItem);
+        assertThat(rootHash).isNotNull();
+        assertThat(rootHash.toString()).isEqualTo(expectedHash);
+    }
+
+    private String testAcceptBlockItem(@NonNull final BlockItem blockItem) throws NoSuchAlgorithmException {
+        // TODO: Not fully implemented yet
+        MerkleNode outputRoot = null;
+        TransactionOutput transactionOutput = null;
+        TransactionResult transactionResult = null;
+        StateChanges stateChanges = null;
+        HashTreeManager<String> tree = new HashTreeManager<>(new SimpleStringCodec());
+        tree.addElement(String.valueOf(blockItem));
+
+        if (blockItem.hasHeader()) {
+            // A header item signals the end of the previous block so we compute the block hash here
+            // May need to call `state.getRootHash()` on the latest `HederaState` here for State Merkle Tree
+
+            // 1. Previous Block Hash Merkle Tree
+            Bytes previousBlockHash = block.items().getFirst().header().previousBlockProofHash();
+
+        } else if (blockItem.hasTransaction()) {
+            // 2. Input Merkle Tree
+            MerkleNode inputRoot = inputRootsByBlock.get(block.items().getFirst());
+            EventMetadata eventMetadata = block.items().get(1).startEventOrThrow();
+            Transaction transaction = block.items().get(1).transactionOrThrow();
+
+        } else if (blockItem.hasTransactionOutput() || blockItem.hasStateChanges()) {
+            // 3. Output Merkle Tree
+            outputRoot = outputRootsByBlock.get(block.items().getFirst());
+            transactionOutput = block.items().get(1).transactionOutputOrThrow();
+            transactionResult = block.items().get(1).transactionResultOrThrow();
+            stateChanges = block.items().get(1).stateChangesOrThrow();
+
+            // 4. State Merkle Tree
+            // todo: manipulate a node from outputRootsByBlock
+        }
+
+        OutputMerkleTreeData outputMerkleTreeData =
+                new OutputMerkleTreeData(outputRoot, transactionOutput, transactionResult, stateChanges);
+        List<OutputMerkleTreeData> outputMerkleTreeDataList = new ArrayList<>();
+        outputMerkleTreeDataList.add(outputMerkleTreeData);
+
+        var rootHash = tree.getTreeRootAsString();
+        return rootHash;
+    }
+}
+
+class SimpleStringCodec implements Codec<String> {
```

Comment:
This class looks like a bunch of stubbed methods. What does it actually do?

Reply:
It's only used for testing purposes by supplying a most basic Codec

Reply:
Ah, there's a default implementation in `Codec` 👍